python/dist2wheel.py

```diff
@@ -73,33 +75,51 @@ def create_all(self):
         for artifact in self.artifacts:
             try:
                 self.path = artifact["path"]
-                self.platform_tag = self._fixup_platform_tag(artifact)
+                self._set_platform_props(artifact)
                 self.checksum = self._fix_checksum(artifact["extra"]["Checksum"])
                 self.size = os.path.getsize(self.path)
             except KeyError:
                 continue
 
             self.wheel_file = WHEEL_TEMPLATE.format(**self.__dict__).encode()
             self.metadata_file = METADATA_TEMPLATE.format(**self.__dict__).encode()
+            os.makedirs(self.wheel_dir, exist_ok=True)
             self._emit()
 
-    @staticmethod
-    def _fixup_platform_tag(artifact):
+    def _matrix(self):
+        return {
+            "darwin": {
+                "amd64": {
+                    "arch": "x86_64",
+                    "platform": "macosx_10_7_x86_64",
+                    "tags": [
+                        f"{self.py_tag}-{self.abi_tag}-macosx_10_7_x86_64",
+                    ],
+                },
+            },
+            "linux": {
+                "amd64": {
+                    "arch": "x86_64",
+                    "platform": "manylinux2014_x86_64.manylinux_2_17_x86_64.musllinux_1_1_x86_64",
+                    "tags": [
+                        f"{self.py_tag}-{self.abi_tag}-manylinux2014_x86_64",
+                        f"{self.py_tag}-{self.abi_tag}-manylinux_2_17_x86_64",
+                        f"{self.py_tag}-{self.abi_tag}-musllinux_1_1_x86_64",
+                    ],
+                },
+            },
+        }
+
+    def _set_platform_props(self, artifact):
         """Convert Go binary nomenclature to Python wheel nomenclature."""
 
-        # Go 1.21 will require macOS 10.15 or later
-        _map = dict(darwin="macosx_10_15", linux="linux")
-        platform = _map[artifact["goos"]]
-
-        arch = artifact["goarch"]
-        if arch == "arm64" and platform == "linux":
-            arch = "aarch64"
-        elif arch == "amd64":
-            arch = "x86_64"
-        elif arch == "386":
-            arch = "i686"
+        _map = self._matrix()
+        _platform = artifact["goos"]
+        _goarch = artifact["goarch"]
 
-        return f"{platform}_{arch}"
+        platform = _map[_platform][_goarch]["platform"]
+        self.platform = f"{platform}"
+        self.platform_tag = "\n".join(_map[_platform][_goarch]["tags"])
```

Comment:
I think you're going to want this to be `"\nTag: ".join(...)` to get the file format correct.

Reply:
thanks, good catch!